src/expandableCalendar/__test__/index.spec.ts

```diff
@@ -0,0 +1,294 @@
+      });
+
+      it(`should call onDateChanged and onMonthChanged for first day in current month when changing to current month`, () => {
+        driver.toggleKnob();
+        jest.runAllTimers();
+        driver.pressOnHeaderArrow({left: false});
+        jest.runAllTimers();
+        driver.pressOnHeaderArrow({left: true});
+        jest.runAllTimers();
+        const expectedDate = today.clone().setDate(1);
+        expect(onDateChanged).toHaveBeenNthCalledWith(2, toMarkingFormat(expectedDate), UpdateSources.PAGE_SCROLL);
+        expect(onMonthChange).toHaveBeenNthCalledWith(2, xdateToData(expectedDate), UpdateSources.PAGE_SCROLL);
+      });
+
+      it(`should navigate 6 months ahead and back successfully`, () => {
+        driver.toggleKnob();
+        jest.runAllTimers();
+        times(6, () => {
+          driver.pressOnHeaderArrow({left: false});
+          jest.runAllTimers();
+        });
+        const expectedFutureDate = today.clone().addMonths(6).setDate(1);
+        expect(onDateChanged).toHaveBeenNthCalledWith(6, toMarkingFormat(expectedFutureDate), UpdateSources.PAGE_SCROLL);
+        expect(onMonthChange).toHaveBeenNthCalledWith(6, xdateToData(expectedFutureDate), UpdateSources.PAGE_SCROLL);
+        times(6, () => driver.pressOnHeaderArrow({left: true}));
+        const expectedDate = today.clone().setDate(1);
+        expect(onDateChanged).toHaveBeenNthCalledWith(12, toMarkingFormat(expectedDate), UpdateSources.PAGE_SCROLL);
+        expect(onMonthChange).toHaveBeenNthCalledWith(12, xdateToData(expectedDate), UpdateSources.PAGE_SCROLL);
+      });
+    });
+  });
+
+  describe('WeekCalendar updates', () => {
+    describe('Day Press', () => {
+      beforeEach(() => {
+        driver.render();
+      });
+      it('should day press update date', () => {
+        driver.selectWeekDay(dashedToday);
+
+        // WeekCalendar
+        const weekCalendarProps = driver.getWeekCalendar().props;
+        const initialScrollIndex = weekCalendarProps.initialScrollIndex;
+        const currentWeek = weekCalendarProps.data[initialScrollIndex];
+        expect(currentWeek).toBe(dashedToday);
+
+        // events
+        expect(onDateChanged).toHaveBeenCalled();
+      });
+    });
+
+    describe('arrow press', () => {
+      beforeEach(() => {
+        driver.render();
+      });
+      it.each([[Direction.LEFT], [Direction.RIGHT]])(`should call onDateChanged to next week first day when pressing %s arrow`, (direction) => {
+        const currentDay = today.getUTCDay();
+        const expectedDate = today.clone().addDays(direction === Direction.LEFT ? - (currentDay + 7) : (7 - currentDay));
+        driver.pressOnHeaderArrow({left: direction === Direction.LEFT});
+        expect(onDateChanged).toHaveBeenCalledWith(toMarkingFormat(expectedDate), UpdateSources.PAGE_SCROLL);
+      });
+
+      it(`should call onDateChanged for first day of current week when changing to current week`, () => {
```

Comment:
Not sure I understand this test... you pressing the right arrow, then the left arrow, and expect the date returned to be the first day of the current week, when the initial date was not the first day of the week. Right?

Reply:
yes, when I initially wrote the test I remembered we returned to today when we scrolled to this week, but after we talked I understood this was not the expected behavior and changed to the correct behavior